src/components/Button/Button.tsx

```diff
@@ -11,21 +11,27 @@ import { TButtonLevel, TButtonSize, TButtonState } from './types';
   Select (-> These might be better served as a different component)
 */
 
+type TagType = {
+  className?: string;
+  [key: string]: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+};
+
 interface BaseButtonProps {
   circle?: boolean;
   className?: string;
   darkMode?: boolean;
   hollow?: boolean;
   level?: TButtonLevel;
   size?: TButtonSize;
-  tag?: React.ComponentType<{ [key: string]: unknown }> | null;
+  tag?: React.FC<TagType> | React.ComponentType<TagType> | null;
 
   // States
   hover?: boolean;
   active?: boolean;
   disabled?: boolean;
   checked?: boolean;
   focus?: boolean;
+  [key: string]: any; // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
TS was being a jerk about letting you pass arbitrary props down when you send a custom tag (like `"div"`, `"a"`, or `MyComponent`).

Reply:
Well, the idea was to limit `<Button>` to be an actual button or anchor tag. `button` and `a` tags, as well as React components, make sense.

What was the problem that required this change? This effectively allows "anything" to be passed and therefore anything that is not a proper HTML attribute will get rendered along with the element. Does that make sense?

Reply:
iirc, we were spreading arbitrary tags onto the Button on purpose.

Not only can you do `aria-label="foo"` (and a number of other arbitrary html attributes), you also can pass a component as a tag, like `tag={NavLink}`, in which case any valid prop for the component is a valid prop for that instance of `Button`.